jhdf/src/test/java/io/jhdf/nio/NioPathTest.java

```diff
@@ -0,0 +1,251 @@
+package io.jhdf.nio;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import io.jhdf.HdfFile;
+import io.jhdf.api.*;
+import io.jhdf.exceptions.HdfException;
+import io.jhdf.object.datatype.DataType;
+
+/**
+ * This test ensures that jHDF supports loading HDF5 files referenced by instances of {@link java.nio.file.Path}
+ * that do not reside in the default file system. To do so, the test
+ * <ol>
+ *     <li>
+ *         copies all test files into a temporary {@code ZipFileSystem} and
+ *     </li>
+ *     <li>
+ *         evaluates that loading the local file and loading the non-local copy yields the same HDF5 structure representation.
+ *     </li>
+ * </ol>
+ */
+class NioPathTest
```

Comment:
Test looks great really thorough. Would it be possible to convert to Hamcrest style `assetThat` assertions. I think there more readable and have generally nicer errors. Not a blocker though can be merged anyway and changed later.

Reply:
I changed the test to Hamcrest style to be consistent with the other tests. Unlike the other tests, I used Matchers.is() as a shortcut for Matchers.is(Matchers.equalTo()). If you find is(equalTo()) more expressive in the code, then I can also use the longer version.